bamboo/test_suite.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+EXIT_CODE=0
+
+function help_message {
+    local SCRIPT=$(basename ${0})
+    local N=$(tput sgr0)
+    local C=$(tput setf 4)
+    cat <<EOF
+Run test configurations of LBANN.
+By default, run all tests.
+Usage: ${SCRIPT} [options]
+Options:
+    ${C}-h|--help${N}                   Display this help message and exit
+    ${C}-u|--unit${N}                   Run all unit tests
+    ${C}-i|--integration${N}            Run all integration tests 
+    ${C}-c|--compiler${N}               Run all compiler tests
+    ${C}-e|--executable${N} <val>       Specify path to executable (default executable built from build_lbann_lc.sh)
+    ${C}-n|--nobuild${N}                Do not build any executable
+EOF
+}
+
+function print_label {
+    TEST_SET=${1}
+    echo "-----------------------------------------------------------------------------------------------------------------"
+    echo "STARTING ${TEST_SET} TESTS"
+    echo "-----------------------------------------------------------------------------------------------------------------"
+}
+
+function print_results {
+    TEST=${1}
+    if [ $? -ne 0 ]; then
+        EXIT_CODE=1
+	echo "-------------------------------------------------------------------------------------------------------------"
+        echo "${TEST} FAILED"
+        echo "-------------------------------------------------------------------------------------------------------------"
+    else
+        echo "-------------------------------------------------------------------------------------------------------------"
+        echo "${TEST} PASSED"
+        echo "-------------------------------------------------------------------------------------------------------------"
+    fi
+}
+
+CLUSTER=$(hostname | sed 's/\([a-zA-Z][a-zA-Z]*\)[0-9]*/\1/g')
+export LBANN_DIR=$(git rev-parse --show-toplevel)
+export LBANN_EXE="${LBANN_DIR}/build/${CLUSTER}.llnl.gov/model_zoo/lbann"
+export PATH=../../LBANN-NIGHTD-CS/bin:$PATH
+
+# Default is to build an executable
+BUILD_EXE=1
+
+# Set defaults to run all tests
+RUN_ALL=1
```

Comment:
Maybe instead of `0` and `1` we should make constants `FALSE` and `TRUE` respectively.